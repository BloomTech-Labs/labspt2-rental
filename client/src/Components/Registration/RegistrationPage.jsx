import React, { Component } from 'react'
import { Button, Form, Divider, Segment, Header } from 'semantic-ui-react'
import { FlexColumn, FlexRow } from 'custom-components'

class RegistrationPage extends Component {
  state = {
    username: '',
    email: '',
    password: ''
  }

  handleInputChange = (event) => {
    const { target: { name, value } } = event
    this.setState({ [name]: value })
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { registerUser } = this.props
    console.log('logging')
    const user = {
      email: this.state.email,
      username: this.state.username,
      password: this.state.password
    }

    event.preventDefault()

    registerUser(user)
  }

  render () {
    return (
      <FlexColumn width="full" alignCenter justifyCenter>
        <Segment className="sm-container">
          <Header size='large'>Registration</Header>

          <Divider/>

          <Form onSubmit={this.handleSubmit}>
            <Form.Field>
              <label htmlFor="username-input">Username</label>
              <input
                id="username-input"
                placeholder='Username'
                name="username"
                value={this.state.username}
                onChange={this.handleInputChange}
              />
            </Form.Field>
            <Form.Field>
              <label htmlFor="email-input">Email</label>
              <input
                id="email-input"
                placeholder='Email'
                name="email"
                value={this.state.email}
                onChange={this.handleInputChange}
              />
            </Form.Field>
            <Form.Field>
              <label htmlFor="password-input">Password</label>
              <input
                id="password-input"
                placeholder='Password'
                name="password"
                type="password"
                value={this.state.password}
                onChange={this.handleInputChange}
              />
            </Form.Field>

            <br/>

            <FlexRow width="full" alignCenter justifyBetween>
              <Button>Login</Button>

              <Button color="green" type='submit'>Submit</Button>
            </FlexRow>
          </Form>
        </Segment>
      </FlexColumn>
    )
  }
}

export default RegistrationPage