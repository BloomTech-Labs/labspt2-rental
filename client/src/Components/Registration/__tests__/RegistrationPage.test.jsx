import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import RegistrationPage from '../RegistrationPage'

afterEach(cleanup)

const registerUser = jest.fn()

describe('RegistrationPage', () => {

  test('calls onSubmit with username, email and password when submitted', () => {
    const fakeUser = {
      username: 'centaur',
      email: 'centaur@centaurs.net',
      password: 'halfhorsehalfman'
    }
  
    const { getByLabelText, getByText } = render(<RegistrationPage registerUser={registerUser} />)
    const usernameNode = getByLabelText('Username')
    const emailNode = getByLabelText('Email')
    const passwordNode = getByLabelText('Password')
    const submitButtonNode = getByText('Submit')
  
    fireEvent.change(usernameNode, { target: { value: fakeUser.username}})
    fireEvent.change(emailNode, { target: { value: fakeUser.email}})
    fireEvent.change(passwordNode, { target: { value: fakeUser.password}})
  
    fireEvent.click(submitButtonNode)
  
    expect(registerUser).toHaveBeenCalledTimes(1)
    expect(registerUser).toHaveBeenCalledWith(fakeUser)
  })
  
  test('snapshot', () => {
    const { container } = render(<RegistrationPage registerUser={registerUser} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  
})
