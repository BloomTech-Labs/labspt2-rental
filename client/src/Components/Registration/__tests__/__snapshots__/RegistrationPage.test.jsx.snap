// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage snapshot 1`] = `
<div
  class="sc-bwzfXH fnoZsX"
  width="full"
>
  <div
    class="ui segment sm-container"
  >
    <div
      class="ui large header"
    >
      Registration
    </div>
    <div
      class="ui divider"
    />
    <form
      class="ui form"
    >
      <div
        class="field"
      >
        <label
          for="username-input"
        >
          Username
        </label>
        <input
          id="username-input"
          name="username"
          placeholder="Username"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <label>
          Email
        </label>
        <input
          id="email-input"
          name="email"
          placeholder="Email"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <label>
          Password
        </label>
        <input
          id="password-input"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <br />
      <div
        class="sc-bdVaJa biWxYW"
        width="full"
      >
        <button
          class="ui button"
        >
          Login
        </button>
        <button
          class="ui green button"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
